### ISSUE TITLE
AUTOPROC ArgumentException: "Can only rename an autoproc if generating one at a time."

### ISSUE THREAD

**Issue by jeroenbu** [jonwagner/Insight.Database.Schema#39]
Hi,

I'm trying to execute this line:
```
AUTOPROC All Gebruikers Name={2}{0} Single=Gebruiker Plural=Gebruikers
```
If I understand the [documentation] (https://github.com/jonwagner/Insight.Database.Schema/wiki/AutoProcs) correctly, this should work: I can use the All keyword with the Name keyword.
The sourcecode however seems to suggest that only one _type can be set at a time. Am I doing something wrong?

Thx,

Jeroen

**Comment by jonwagner** [comment]
That's a current limitation.

This doesn't work:

* Install AUTOPROC All Name={0}
* Upgrade to AUTOPROC All Name={0}_Other

But this does work:

* Install AUTOPROC All Name={0}
* Remove the autoprocs
* Upgrade to AUTOPROC All Name={0}_Other

Most people only run into this issue during development, so I haven't taken the time to fix it.

**Comment by jeroenbu** [comment]
Hm, strange. I did just that (remove the autoprocs, first only the stored procedures, and next also the types), but the error is still thrown.

Got any more suggestions? And, how should I configure AUTOPROC so it will deploy correctly with Name={2}_{0} (or Name={2}{0}) in production?

Thanks,

Jeroen

**Comment by jonwagner** [comment]
If you removed them manually, then you also want to remove the records from the Insight_SchemaRegistry table. (That's how Insight tracks changes.)

Another option is to run the install with the rebuild option. It will drop everything other than the data (procs, types, views, indexes) and reinstall everything.

**Comment by jeroenbu** [comment]
Actually, I tried that.

1. when I delete all autoproc objects (table Insight_SchemaRegistry, generated procs, generated datatypes), and I try to have 
```
AUTOPROC All Gebruikers Name={2}_{0} Single=Gebruiker Plural=Gebruikers
```
executed, it gives me the error.
2. when i first execute the AUTOPROC command without the Name-keyword, then delete the autoprocs and the autoproc record from the SchemaRegistry, then add Name={2}_{0} in the AUTOPROC command, I get the error again when trying to execute the command.

And, if I read the sourcecode correctly, it checks if more than one type of autoproc is being (re)generated, and it will throw the error if that is the case.

**Comment by jonwagner** [comment]
I checked into the code. That exception was a limitation of the library, but I have a fix for that.

 

I think I’ve resolved all of your issues. I can either post the code for you to try out, or you can send me your SQL schema and I can run it through the verifier before I do a build.

**Comment by jonwagner** [comment]
I pushed a code fix for this. If you want, you can email me your SQL script and I'll run it through the verifier and fix any issues before I do an official build.

**Comment by jonwagner** [comment]
My hotmail account is jonwagner

**Comment by jonwagner** [comment]
This is fixed in v2.2.10
